### PR TITLE
Adding Higra python package

### PR DESCRIPTION
**Pull request by JoOkuma**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by JoOkuma** [comment]
Hi, @PerretB. Could you please confirm you want to be a maintainer of this recipe?

**Comment by JoOkuma** [comment]
Hi, @conda-forge/help-python-c. Could you provide some input on how to fix this issue?

`win` build running [out of memory](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=935850&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4)
```bash
  MSBUILD : error :     System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown. [C:\bld\higra_1715884739591\work\build\temp.win-amd64-cpython-312\Release\higra\higram.vcxproj]
  MSBUILD : error :    at Go6(RegexRunner ) [C:\bld\higra_1715884739591\work\build\temp.win-amd64-cpython-312\Release\higra\higram.vcxproj]

```

File: recipes/higra/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - osx-arm64                  # [osx]
+  - linux-64                   # [linux and x86_64]
+  - linux-aarch64              # [linux and not x86_64]
+  - win-64                     # [win]
+
+channel_sources:
+  - conda-forge,defaults
+
+channel_targets:
+  - conda-forge main
```
Comment (xhochy) [inline recipes/higra/conda_build_config.yaml:21]: You should not have this in the file here.
```suggestion
```

File: recipes/higra/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+c_stdlib:
+  - sysroot                    # [linux]
+  - macosx_deployment_target   # [osx]
+  - vs                         # [win]
+
```
Comment (xhochy) [inline recipes/higra/conda_build_config.yaml:5]:
```suggestion
```

File: recipes/higra/conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+c_stdlib_version:              # [unix]
+  - 2.17                       # [linux]
+  - 11.0                       # [osx]
```
Comment (xhochy) [inline recipes/higra/conda_build_config.yaml:8]: Do you really need the last line?

File: recipes/higra/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+{% set version = "0.6.8" %}
+{% set sha256 = "4b86b31bac16c216048f29c5ddd7660a0712770d1de055f9825514fa4d64a2c6" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/higra/Higra/archive/refs/tags/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (xhochy) [inline recipes/higra/meta.yaml:11]: Please don't add this indirection, add the `sha256` directly here.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/higra/conda_build_config.yaml (hunk 3 above)
Comment (JoOkuma) [reply recipes/higra/conda_build_config.yaml:8]: I'm not really sure. I'll try removing it.

File: recipes/higra/meta.yaml (hunk 4 above)
Comment (JoOkuma) [reply recipes/higra/meta.yaml]: Done!

File: recipes/higra/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - ninja  # [win]
+  host:
+    - python
+    - pip
+    - numpy
+  run:
+    - python
+    - numpy
+
+test:
```
Comment (xhochy) [inline recipes/higra/meta.yaml:32]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by JoOkuma** [comment]
thanks a lot @xhochy, everything seems to be OK now

**Review by xhochy — APPROVED** [review]

**Pull request merged**